src/UglyToad.PdfPig/Util/Matrix3x3.cs

```diff
@@ -69,14 +69,14 @@ public IEnumerator<double> GetEnumerator()
         ///
         /// If an inverse matrix does not exist, null is returned.
         /// </summary>
-        public Matrix3x3? Inverse()
+        public Matrix3x3 Inverse()
         {
             var determinant = GetDeterminant();
 
             // No inverse matrix exists when determinant is zero
             if (determinant == 0)
             {
-                return null;
+                throw new InvalidOperationException("May not inverse a matrix with a determinant of 0.");

```

Comment:
NOTE: our only caller will throw an NRE if this method returns null. Throwing early should give us a better debugging experience.